src/services/parserContext.services.ts

```diff
@@ -48,29 +48,33 @@ spdxJsonParser = function (sbom): Package[] {
   sbomPackages.forEach(function (pkg: any) {
     let p: Package = {
       name: pkg.name,
-      id: pkg.SPDXID,
-      purl: undefined,
-      cpeName: undefined,
-      impact: undefined,
-      consRisk: undefined,
-      highestRisk: undefined,
-      likelihood: undefined,
-      version: pkg.versionInfo,
+      ref: pkg.SPDXID,
+      purl: null,
+      cpeName: null,
+      impact: null,
+      consRisk: null,
+      highestRisk: null,
+      likelihood: null,
+      version: null,
     };
 
     //Get External Reference locators
     if (pkg.hasOwnProperty('externalRefs')) {
       for (let extRef of pkg.externalRefs) {
-        if (extRef.referenceType == 'purl' && p.purl == undefined) {
+        if (extRef.referenceType == 'purl' && p.purl == null) {
           p.purl = extRef.referenceLocator;
           break; //stop if EITHER cpe or purl found
         }
-        if (extRef.referenceType.includes('cpe') && p.cpeName == undefined) {
+        if (extRef.referenceType.includes('cpe') && p.cpeName == null) {
           p.cpeName = extRef.referenceLocator;
           break; //stop if EITHER cpe or purl found
         }
       }
     }
+    //Get Version
```

Comment:
why does package version have its own if statement? Does it usually have more than one version? Ik you're doing this for external ref because there's many but provide an explaination here as to why

Reply:
if you read the SBOM specs, version is optional